From 47631f07d19ad73a526afe8e3d0f1d11e7ac3d1c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matthias=20Mail=C3=A4nder?=
 <Mailaender@users.noreply.github.com>
Date: Sat, 5 Aug 2017 20:14:14 +0200
Subject: [PATCH 1/3] Create freeorion.appdata.xml

which adds metadata for Linux software galleries.
---
 freeorion.appdata.xml | 46 ++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 46 insertions(+)
 create mode 100644 freeorion.appdata.xml

diff --git a/freeorion.appdata.xml b/freeorion.appdata.xml
new file mode 100644
index 0000000000..146a794fe0
--- /dev/null
+++ b/freeorion.appdata.xml
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>freeorion.desktop</id>
+  <name>FreeOrion</name>
+  <summary>Turn-based space empire and galactic conquest (4X) computer game</summary>
+  <summary xml:lang="de">Rundenbasierendes Weltraum-Strategiespiel</summary>
+  <developer_name>FreeOrion Project</developer_name>
+  <description>
+  <p>
+   FreeOrion is a free, open source, turn-based space empire and galactic conquest (4X) computer game being designed and built by the FreeOrion project. FreeOrion is inspired by the tradition of the Master of Orion games, but is not a clone or remake of that series or any other game.
+  </p>
+  </description>
+  <keywords>
+    <keyword>game</keyword>
+    <keyword>strategy</keyword>
+  </keywords>
+  <metadata_license>CC-BY-SA-3.0</metadata_license>
+  <project_license>GPL-2.0</project_license>
+  <url type="bugtracker">https://github.com/freeorion/freeorion/issues</url>
+  <url type="donation">http://www.freeorion.org/index.php/Donations</url>
+  <url type="faq">http://www.freeorion.org/index.php/FAQ</url>
+  <url type="help">http://www.freeorion.org/forum/viewforum.php?f=25</url>
+  <url type="homepage">http://www.freeorion.org</url>
+  <screenshots>
+    <screenshot type="default">
+      <caption>Galaxy setup</caption>
+      <image height="600" width="786">http://www.freeorion.org/images/thumb/1/1a/Freeorion_0.4.5_b2015-09-01.f203162_singleplayer.JPG/786px-Freeorion_0.4.5_b2015-09-01.f203162_singleplayer.JPG</image>
+    </screenshot>
+<screenshot type="default">
+      <caption>Ship design</caption>
+      <image height="599" width="787">http://www.freeorion.org/images/thumb/c/c7/Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_8.JPG/787px-Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_8.JPG</image>
+    </screenshot>
+    <screenshot type="default">
+      <caption>Map exploration</caption>
+      <image height="599" width="787">http://www.freeorion.org/images/thumb/e/ee/Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_7.JPG/787px-Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_7.JPG</image>
+    </screenshot>
+    <screenshot type="default">
+      <caption>In game windows</caption>
+      <image height="599" width="786">http://www.freeorion.org/images/thumb/0/06/Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_2.JPG/786px-Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_2.JPG</image>
+    </screenshot>
+    <screenshot type="default">
+      <caption>In game windows</caption>
+      <image height="599" width="788">http://www.freeorion.org/images/thumb/6/62/Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_1.JPG/788px-Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_1.JPG</image>
+    </screenshot>
+  </screenshots>
+</component>

From ab6ca60e61c0baab18fef61ef1142c08a072c77e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matthias=20Mail=C3=A4nder?=
 <Mailaender@users.noreply.github.com>
Date: Sat, 5 Aug 2017 20:15:06 +0200
Subject: [PATCH 2/3] Install the freeorion.appdata.xml

---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 448c35660b..b8363a07b0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -742,6 +742,12 @@ if(UNIX AND NOT APPLE)
         PATTERN "*~" EXCLUDE
         PATTERN "*.pyc" EXCLUDE
     )
+    
+    install(
+        FILES
+        ${CMAKE_SOURCE_DIR}/freeorion.appdata.xml
+        DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/appdata
+    )
 
     foreach(SIZE 16 24 32 64 128 256)
         install(

From 87b7837ab876f8e138b6f31dfddcf99e16750c76 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matthias=20Mail=C3=A4nder?=
 <Mailaender@users.noreply.github.com>
Date: Sat, 5 Aug 2017 20:17:58 +0200
Subject: [PATCH 3/3] Update freeorion.appdata.xml

I assume type="default" should only be used once.
---
 freeorion.appdata.xml | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/freeorion.appdata.xml b/freeorion.appdata.xml
index 146a794fe0..5af801fa34 100644
--- a/freeorion.appdata.xml
+++ b/freeorion.appdata.xml
@@ -26,19 +26,19 @@
       <caption>Galaxy setup</caption>
       <image height="600" width="786">http://www.freeorion.org/images/thumb/1/1a/Freeorion_0.4.5_b2015-09-01.f203162_singleplayer.JPG/786px-Freeorion_0.4.5_b2015-09-01.f203162_singleplayer.JPG</image>
     </screenshot>
-<screenshot type="default">
+    <screenshot>
       <caption>Ship design</caption>
       <image height="599" width="787">http://www.freeorion.org/images/thumb/c/c7/Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_8.JPG/787px-Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_8.JPG</image>
     </screenshot>
-    <screenshot type="default">
+    <screenshot>
       <caption>Map exploration</caption>
       <image height="599" width="787">http://www.freeorion.org/images/thumb/e/ee/Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_7.JPG/787px-Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_7.JPG</image>
     </screenshot>
-    <screenshot type="default">
+    <screenshot>
       <caption>In game windows</caption>
       <image height="599" width="786">http://www.freeorion.org/images/thumb/0/06/Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_2.JPG/786px-Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_2.JPG</image>
     </screenshot>
-    <screenshot type="default">
+    <screenshot>
       <caption>In game windows</caption>
       <image height="599" width="788">http://www.freeorion.org/images/thumb/6/62/Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_1.JPG/788px-Freeorion_0.4.5_b2015-09-01.f203162_ingame-windows_1.JPG</image>
     </screenshot>
